### PR TITLE
Gives the uploader the ability to add fields to the tiddler

### PR DESCRIPTION
**Pull request by FSpark**
In order to facilitate adding some meta information to the uploaded files.

### CODE REVIEW

File: plugins/file-uploads/upload-handler.js [hunk 1]
```diff
@@ -176,9 +176,13 @@ UploadTask.prototype.changeCountUnchanged = function(title) {
 // - the tiddler has not changed since we uploaded it
 UploadTask.prototype.makeCanonicalURITiddler = function(title) {
 	var tiddler = this.wiki.getTiddler(title),
-		canonical_uri = this.tiddlerInfo[title].canonical_uri;
+		canonical_uri = this.tiddlerInfo[title].canonical_uri,
+		fields = this.tiddlerInfo[title].fields;
+	
+	var options = Object.assign({},{text:"",_canonical_uri:canonical_uri},fields);
```
Comment (saqimtiaz) [inline plugins/file-uploads/upload-handler.js]: @FSpark You can skip assigning the `options` variable here and instead in line 185 just pass as follows to `new  $tw.Tiddler`:

`this.wiki.addTiddler(new $tw.Tiddler(tiddler,{text:"",_canonical_uri:canonical_uri},fields));`

**Comment by saqimtiaz** [comment]
@FSpark thank you. I have requested a small change otherwise this looks like a good improvement.

What backend are you writing an uploader module for?

File: plugins/file-uploads/upload-handler.js (hunk 1 above)
Comment (FSpark) [reply plugins/file-uploads/upload-handler.js]: Thanks for your suggestion, it was indeed redundant and now fixed.

**Comment by FSpark** [comment]
@saqimtiaz [AliCloud OSS](https://www.alibabacloud.com/en/product/object-storage-service) and [sm.ms](https://sm.ms), an image hosting service, which generate a hash for each image for subsequent manipulation of that image.  I would like to implement operations such as deletion for uploaded files.

**Pull request merged**